### PR TITLE
feat(core): notification mobile (in popover)

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue(s)

Closes https://github.com/SAP/fundamental-ngx/issues/7965

## Description

Notification mobile mode (in Popover).

### After:

image

image

### CODE REVIEW

File: libs/core/src/lib/notification/notification-group/notification-group.component.ts [hunk 1]
```diff
@@ -30,6 +30,10 @@ export class NotificationGroupComponent implements OnChanges, OnInit, CssClassBu
     @Input()
     width: string;
 
+    /** @hidden */
+    @HostBinding('style.height')
+    _height = '100%';
```
Comment (mikerodonnell89) [inline libs/core/src/lib/notification/notification-group/notification-group.component.ts:35]: Why as a hostbinding and not just styled in the css (in fundamental-styles)?

**Review by mikerodonnell89 — COMMENTED** [review]
Looks good, did have one question see above

File: libs/core/src/lib/notification/notification-group/notification-group.component.ts (hunk 1 above)
Comment (platon-rov) [reply libs/core/src/lib/notification/notification-group/notification-group.component.ts:35]: moved https://github.com/SAP/fundamental-styles/pull/3783

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
I find this odd. If I see this in an app I'd immediately assume it's a bug. First of all, the specs say the following: "Notification is carried inside a Popover. Popover can be responsive (on mobile become Dialog) **or not.**"

The `or not` part means we need to have this behaviour as a property so the user can specify if they want the notification to appear in a dialog or keep the normal popover behaviour. 

Second, the box-shadow of the notification that creates the illusion of a border should probably be removed when it's in a responsive popover and I don't see the need of a footer in this case. 

Screen Shot 2022-08-10 at 3 08 52 PM
Screen Shot 2022-08-10 at 3 08 23 PM

**Comment by platon-rov** [comment]
> I find this odd. If I see this in an app I'd immediately assume it's a bug. First of all, the specs say the following: "Notification is carried inside a Popover. Popover can be responsive (on mobile become Dialog) **or not.**"
> The `or not` part means we need to have this behaviour as a property so the user can specify if they want the notification to appear in a dialog or keep the normal popover behaviour.

Having popover mobile so much different from the normal popover is the architectural mistake, i would prefer to fix it in the popover component, otherwise any component where we want to use both - normal & mobile popover will look very odd.

More info: https://github.com/SAP/fundamental-ngx/issues/8574

> Second, the box-shadow of the notification that creates the illusion of a border should probably be removed when it's in a responsive popover and I don't see the need of a footer in this case.

Fixed.

File: libs/docs/core/notification/examples/notification-mobile/notification-mobile-example.component.html [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    </ng-template>
+</fd-popover>
+
+<h4>Notification Group In Popover Dialog</h4>
+
+<button fd-button label="Notification Group" fdCompact [fdPopoverTrigger]="popoverMobileGroup"></button>
+
+<fd-popover #popoverMobileGroup [mobile]="true" [mobileConfig]="mobileConfig">
+    <ng-template #popoverBodyContent>
+        <fd-notification-group width="100%">
```
Comment (mikerodonnell89) [inline libs/docs/core/notification/examples/notification-mobile/notification-mobile-example.component.html:116]: I don't think `width="100%"` is needed

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Example code is not displaying

Screen Shot 2022-10-04 at 1 59 24 PM

File: libs/core/src/lib/dialog/dialog.component.scss [hunk 3]
```diff
@@ -22,4 +22,9 @@
             border-top-right-radius: 0;
         }
     }
+
+    .fd-dialog__body--no-horizontal-padding {
```
Comment (mikerodonnell89) [inline libs/core/src/lib/dialog/dialog.component.scss:26]: I think this should be in fundamental-styles, oddly it is mentioned in the documentation there but this class is missing

File: libs/docs/core/notification/examples/notification-mobile/notification-mobile-example.component.html [hunk 4]
```diff
@@ -0,0 +1,263 @@
+<h4>Notification In Popover Dialog</h4>
+
+<button fd-button label="Notification" fdCompact [fdPopoverTrigger]="popoverMobile"></button>
```
Comment (mikerodonnell89) [inline libs/docs/core/notification/examples/notification-mobile/notification-mobile-example.component.html:3]: Some buttons are hardcoded compact and some respond to the content density observer. Is this intentional? Looks like in the specs they should all be compact

**Review by mikerodonnell89 — APPROVED** [review]
Approving but I do have two comments/questions, see above

**Comment by InnaAtanasova** [comment]
what is the black border around the dialog? And why aren't the bottom corners rounded?
Screen Shot 2022-10-24 at 1 26 45 PM
Screen Shot 2022-10-24 at 1 26 50 PM

**Comment by InnaAtanasova** [comment]
I find this very weird: 
Screen Shot 2022-10-24 at 1 27 52 PM
Shouldn't the elements have the same width?

**Comment by InnaAtanasova** [comment]
Clicking outside of the dialog and ESC should close the notifications "dialog"

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
See my comments

File: libs/docs/core/notification/examples/notification-mobile/notification-mobile-example.component.html (hunk 4 above)
Comment (platon-rov) [reply libs/docs/core/notification/examples/notification-mobile/notification-mobile-example.component.html:3]: Good catch, fixed for all examples

File: libs/core/src/lib/dialog/dialog.component.scss (hunk 3 above)
Comment (platon-rov) [reply libs/core/src/lib/dialog/dialog.component.scss:26]: https://github.com/SAP/fundamental-styles/pull/3957

**Comment by platon-rov** [comment]
> what is the black border around the dialog? 

Cannot reproduce, chrome/firefox.

> And why aren't the bottom corners rounded?

In mobile popover (i.e. full width dialog) it makes sense to have all the borders angled so i made exact this https://github.com/SAP/fundamental-styles/pull/3957

> Shouldn't the elements have the same width?

I think mobile popover may be 100vh and 100vw but this one of many questions to this component, let's maybe fix it separately? Added to this ticket https://github.com/SAP/fundamental-ngx/issues/8574

> Clicking outside of the dialog and ESC should close the notifications "dialog"

Mobile popover doesn't have such behavior opposite to the Dialog. I also described it there https://github.com/SAP/fundamental-ngx/issues/8574

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**